CONTRIBUTING.md

```diff
@@ -0,0 +1,126 @@
+
+Ensure you have the following installed in your machine:
+
+- [Java Development Kit](https://docs.oracle.com/javase/8/docs/technotes/guides/install/install_overview.html)
+- [Android Studio and Android SDK](https://developer.android.com/studio/index.html) check [build.gradle](./app/build.gradle) file for the required SDK versions and build tools version
+- [Git SCM](http://git-scm.com/)
+
+## Cloning the repository
+
+```bash
+git clone git@github.com:aerogear/android-showcase-template.git
+cd aerogear-showcase-template/
+```
+
+## Installing dependencies and building the ShowCase Template
+
+To run all the tests:
+```bash
+./gradlew test connectedAndroidTest
+```
+To build:
+```bash
+/gradlew assembleDebug
+```
+To do release build (you will need to setup the signing credential):
+```bash
+./gradlew aR
+```
+See the [Gradle Documentation](https://docs.gradle.org/current/userguide/pt02.html) for more information on Gradle and the Gradle Wrapper
+
+## Using a Published SDK
+
+### jcenter()
+
+To use a published version of the SDK, add jcenter to the [build.gradle](https://github.com/aerogear/android-showcase-template/blob/master/build.gradle) file in the root directory of this project.
+
+```groovy
+allprojects {
+    repositories {
+        jcenter() // <-- Add This line
+        google()
+    }
+}
+```
+
+### Reference the Dependencies
+
+In the [build.gradle](./app/build.gradle) file in the app directory, add the dependencies provided by the SDK.
+
+```groovy
+dependencies {
+    ...
+    implementation 'org.aerogear:core:0.1.0-2018-09'
+    implementation 'org.aerogear:auth:0.1.0-2018-09' // <- adjust to suit release version
+}
+```
+
+A full list of releases can be viewed [here](https://github.com/aerogear/aerogear-android-sdk/releases).
+
+## Installing Local Build of SDK
+
+Alternatively the SDK can be installed manually instead of using a published version.
+
+### Clone and build the library
+
+```bash
+git clone https://github.com/aerogear/aerogear-android-sdk
+cd aerogear-android-sdk
+./gradlew install
+```
+
+It's possible to verify the install went well by checking the local maven repository:
+
+```bash
+ls ~/.m2/repository/org/aerogear
+# Should output android-push android-core android-auth 
+```
+
+### MavenLocal()
+
+If the build of the SDK is from source, add mavenLocal() to the [build.gradle](https://github.com/aerogear/android-showcase-template/blob/master/build.gradle) file in the root directory of this project.
+
+```groovy
+allprojects {
+    repositories {
+        mavenLocal() // <-- Add This line
+        google()
+        jcenter()
+    }
+}
+```
+
+### Reference the Dependencies
+
+In the [build.gradle](./app/build.gradle) file in the app directory, add the dependencies provided by the SDK.
+
+```groovy
+dependencies {
+    ...
+    implementation 'org.aerogear:android-core:5.0.0-SNAPSHOT'
+    implementation 'org.aerogear:android-auth:5.0.0-SNAPSHOT' //If you are using IDM services provided by KeyCloak
```

Comment:
Same as I've previously commented.

Reply:
Good catch, think I will revisit docs in the sdk and make sure they follow the same convention

Reply:
@ciaranRoche I already did that in all Android SDK docs ;)